client/src/pages/Mates/MatesPage.tsx

```diff
@@ -1,12 +1,325 @@
+  },
+  {
+    id: id++,
+    ts: new Date("2021-04-09"),
+    title: "Coffee at Starbucks",
+    details:
+      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
+    syncStatus: SyncStatusEnum.COMPLETED,
+  },
+  {
+    id: id++,
+    ts: new Date("2021-06-19"),
+    title: "Snoozed One",
+    details:
+      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
+    syncStatus: SyncStatusEnum.SNOOZED,
+  },
+  {
+    id: id++,
+    ts: new Date("2021-07-15"),
+    title: "Busy with Work",
+    details: "",
+    syncStatus: SyncStatusEnum.DECLINED,
+  },
+  {
+    id: id++,
+    ts: new Date("2021-07-21"),
+    title: "Hiking at Stone Mountain",
+    details:
+      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
+    syncStatus: SyncStatusEnum.COMPLETED,
+  },
+  {
+    id: id++,
+    ts: new Date("2021-08-31"),
+    title: "Flaky?",
+    details:
+      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
+    syncStatus: SyncStatusEnum.DECLINED,
+  },
+];
+
+const newSync = {
+  id: id++,
+  ts: new Date(),
+  title: "New Title",
+  details: "Details here...",
+  syncStatus: SyncStatusEnum.DECLINED,
+};
+/* Hardcoded data above */
+
+const { confirm } = Modal;
 const { Title } = Typography;
 
-const MatesPage = (): JSX.Element => (
-  <>
-    <Title>Mates!</Title>
-  </>
-);
+/**
+ * MatePage
+ * @returns
+ */
+const MatesPage = (): JSX.Element => {
+  const [state, setState] = useState<MatesPageState>({
+    syncs: initialSyncs,
+    isModalVisible: false,
+    editingSync: newSync,
+  });
+  const { syncs, isModalVisible, editingSync } = state;
+
+  /**
+   * Removes specified sync
+   * @param sync
+   */
+  const removeSync = (sync: SyncType) => {
+    confirm({
+      title: `Are you sure you want to delete this sync: ${sync.title}?`,
+      icon: <ExclamationCircleOutlined />,
+      content: "This operation cannot be reversed",
+      okText: "Delete",
+      okType: "danger",
+      cancelText: "No",
+      onOk() {
+        // TODO server
+        setState({
+          ...state,
+          syncs: state.syncs.filter((s) => s.id !== sync.id),
+        });
+      },
+    });
+  };
+
+  /**
+   * Open modal to edit specified sync
+   * @param sync
+   */
+  const editSync = (sync: SyncType) => {
+    setState({ ...state, isModalVisible: true, editingSync: { ...sync } });
+  };
+
+  /**
+   * Add a new sync then edit
+   */
+  const addSync = () => {
+    editSync({
+      id: id++,
+      title: "",
+      details: "",
+      ts: new Date(),
+      syncStatus: SyncStatusEnum.COMPLETED,
+    });
+  };
+
+  /**
+   * Handle save on editing sync
+   * TODO: save on server
+   */
+  const handleOk = () => {
+    const newSyncs = [
```

Comment:
No need to actually pursue this, but the useReducer hook is something I found out about recently - you can dispatch events like ("remove", s.id) or ("update", {s.id,newSync})
It would consolidate the array manipulation logic in one place, but may inadvertently increase code complexity.

Reply:
Good advice. I think the current logic flow is simple enough to leave as it is. Perhaps if it gets more complex when we connect to backend we can revisit the idea?